deployments/gpu-operator/reconfigure-mig.go

```diff
@@ -661,97 +481,40 @@ func hostStopSystemdServices(hostRootMount string, hostGPUClientServices []strin
-		if err = cmd.Run(); err != nil {
-			fmt.Printf("Skipping %s (no-exist)", s)
-			continue
-		}
-
-		cmd = exec.Command("chroot", hostRootMount, "systemctl", "-q", "is-failed", s)
-		if err = cmd.Run(); err == nil {
-			fmt.Printf("Skipping %s (is-failed, will-restart)\n", s)
-			hostGPUClientServicesStopped = append([]string{s}, hostGPUClientServicesStopped...)
-			continue
-		}
-
-		cmd = exec.Command("chroot", hostRootMount, "systemctl", "-q", "is-enabled", s)
-		if err = cmd.Run(); err != nil {
-			fmt.Printf("Skipping %s (disabled)\n", s)
-			continue
-		}
-
-		cmd = exec.Command("chroot", hostRootMount, "systemctl", "show", "--property=Type", s)
-		output, err := cmd.Output()
-		if err != nil {
-			return fmt.Errorf("failed to get service type for %s: %w", s, err)
-		}
-		if strings.TrimSpace(string(output)) == "Type=oneshot" {
-			fmt.Printf("Skipping %s (inactive, oneshot, no-restart)\n", s)
-			continue
-		}
-
-		fmt.Printf("Skipping %s (inactive, will-restart)", s)
-		hostGPUClientServicesStopped = append([]string{s}, hostGPUClientServicesStopped...)
+		hostStopSystemdServicesInactive(hostRootMount, hostGPUClientServices, hostGPUClientServicesStopped, s)
 	}
 	return nil
 }
 
-/*
-applyMIGModeChange
-echo "Applying the MIG mode change from the selected config to the node (and double checking it took effect)"
-echo "If the -r option was passed, the node will be automatically rebooted if this is not successful"
-nvidia-mig-parted -d apply --mode-only -f ${MIG_CONFIG_FILE} -c ${SELECTED_MIG_CONFIG}
-nvidia-mig-parted -d assert --mode-only -f ${MIG_CONFIG_FILE} -c ${SELECTED_MIG_CONFIG}
-if [ "${?}" != "0" ] && [ "${WITH_REBOOT}" = "true" ]; then
-
-	echo "Changing the 'nvidia.com/mig.config.state' node label to 'rebooting'"
-	kubectl label --overwrite  \
-		node ${NODE_NAME} \
-		nvidia.com/mig.config.state="rebooting"
-	if [ "${?}" != "0" ]; then
-		echo "Unable to set the value of 'nvidia.com/mig.config.state' to 'rebooting'"
-		echo "Exiting so as not to reboot multiple times unexpectedly"
-		exit_failed
-	fi
-	chroot ${HOST_ROOT_MOUNT} reboot
-	exit 0
-
-fi
-*/
-func applyMIGModeChange(nvidiaMigPartedAlias string, migConfigFile string, selectedMIGConfig string,
-	 nodeName string, withReboot bool, hostRootMount string, currentLabels map[string]string, hostGPUClientServices []string, withShutdownHostGPUClients bool) error {
-	applyCmd := exec.Command(nvidiaMigPartedAlias, "-d", "apply", "--mode-only", "-f", migConfigFile, "-c", selectedMIGConfig)
-	assertCmd := exec.Command(nvidiaMigPartedAlias, "-d", "assert", "--mode-only", "-f", migConfigFile, "-c", selectedMIGConfig)
+// Applying the MIG mode change from the selected config to the node (and double checking it took effect)
+func applyMIGModeChange(migConfigProperties migConfig, selectedMIGConfig string,
+	nodeName string, withReboot bool, hostRootMount string, currentLabels map[string]string, hostGPUClientServices []string, withShutdownHostGPUClients bool) error {
```

Comment:
Instead of passing all the parameters required to set the sate here, let's return a `rebootRequired` boolean value instead and respond to this in the `reconfigure` function. This will further clean up where we call `setState`.

So we would have:
```suggestion
func applyMIGModeChange(migConfigProperties migConfig, selectedMIGConfig string) (bool, error) {
```